Sources/Swift/Integrations/SessionReplay/SentrySessionReplay.swift

```diff
@@ -248,6 +248,7 @@ class SentrySessionReplay: NSObject {
     private func captureSegment(segment: Int, video: SentryVideoInfo, replayId: SentryId, replayType: SentryReplayType) {
         let replayEvent = SentryReplayEvent(eventId: replayId, replayStartTimestamp: video.start, replayType: replayType, segmentId: segment)
         
+        replayEvent.sdk = self.replayOptions.sdkInfo
```

Comment:
we should also set the same sdkInfo to the envelope header, which this replay event is a part of. And from what I see we don't set this `sdkInfo` to the default global one when initialized, which means we'll nullify the sdk context if we're not running on a hybrid SDK, right?  @philprime @philipphofmann could you guys do that or should I take care while Dhiogo is out?

Reply:
Not sure if I know how to do that yet, would be great if you or @philipphofmann can do it, with me reviewing the changes so I know afterwards.

Reply:
I'm confused by this PR; see https://github.com/getsentry/sentry-cocoa/pull/4663#pullrequestreview-2527610716.

Reply:
I will try to align this PR with how we do this in Java:
* https://github.com/getsentry/sentry-java/pull/4014/files#diff-368c74c6e2f2e3bc57f8d51fea2db766c2455714eba6d4fe40656cbfc1b50855
* https://github.com/getsentry/sentry-java/pull/4015/files#diff-c3475688f47a9611d7c966a91d999cc7c1d642b9bb16cd3223aed5722ec0608bR38-R42

Reply:
I think Java is wrong in this case.
The nativeSDKName and nativeSDKVersion in the replay tags should never be overwritten.
@romtsn Am I missing something here?